src/HonzaBotner.Discord.Services/EventHandlers/VerificationEventHandler.cs

```diff
@@ -9,45 +11,73 @@
 
 namespace HonzaBotner.Discord.Services.EventHandlers
 {
-    public class VerificationEventHandler : IEventHandler<MessageReactionAddEventArgs>
+    public class VerificationEventHandler : IEventHandler<ComponentInteractionCreateEventArgs>
     {
-        private readonly IAuthorizationService _authorizationService;
         private readonly IUrlProvider _urlProvider;
-        private readonly CommonCommandOptions _config;
+        private readonly ButtonOptions _buttonOptions;
+        private DiscordRoleConfig _discordRoleConfig;
 
-        public VerificationEventHandler(IAuthorizationService authorizationService, IUrlProvider urlProvider,
-            IOptions<CommonCommandOptions> options)
+        public VerificationEventHandler(
+            IUrlProvider urlProvider,
+            IOptions<ButtonOptions> options,
+            IOptions<DiscordRoleConfig> discordRoleConfig
+        )
         {
-            _authorizationService = authorizationService;
             _urlProvider = urlProvider;
-            _config = options.Value;
+            _buttonOptions = options.Value;
+            _discordRoleConfig = discordRoleConfig.Value;
         }
 
-        public async Task<EventHandlerResult> Handle(MessageReactionAddEventArgs eventArgs)
+        public async Task<EventHandlerResult> Handle(ComponentInteractionCreateEventArgs eventArgs)
         {
-            // https://discordapp.com/channels/366970031445377024/507515506073403402/686745124885364770
+            if (eventArgs.Id != _buttonOptions.VerificationId) return EventHandlerResult.Continue;
 
-            if (!(eventArgs.Message.Id == _config.VerificationMessageId
-                  && eventArgs.Message.ChannelId == _config.VerificationChannelId))
-                return EventHandlerResult.Continue;
-            if (!eventArgs.Emoji.Name.Equals(_config.VerificationEmojiName)) return EventHandlerResult.Continue;
+            DiscordInteractionResponseBuilder builder = new DiscordInteractionResponseBuilder().AsEphemeral(true);
 
             DiscordUser user = eventArgs.User;
-            DiscordDmChannel channel = await eventArgs.Guild.Members[user.Id].CreateDmChannelAsync();
+            DiscordMember member = eventArgs.Guild.Members[user.Id];
 
-            string link = _urlProvider.GetAuthLink(user.Id, RolesPool.Auth);
+            string link = _urlProvider.GetAuthLink(eventArgs.User.Id, RolesPool.Auth);
```

Comment:
Nit: Since you have taken out user to property, you can use it.
```suggestion
            string link = _urlProvider.GetAuthLink(user.Id, RolesPool.Auth);
```